snowflake-api/src/session.rs

```diff
@@ -217,18 +239,22 @@ impl Session {
                 .is_some_and(|at| at.master_token.is_expired())
         {
             // Create new session if tokens are absent or can not be exchange
-            let tokens = match self.auth_type {
-                AuthType::Certificate => {
+            let tokens = match &self.auth_type {
+                AuthType::Certificate(pem) => {
                     log::info!("Starting session with certificate authentication");
                     if cfg!(feature = "cert-auth") {
-                        self.create(self.cert_request_body()?).await
+                        self.create(self.cert_request_body(pem)?).await
                     } else {
                         Err(AuthError::MissingCertificate)?
                     }
                 }
-                AuthType::Password => {
+                AuthType::Password(pwd) => {
                     log::info!("Starting session with password authentication");
-                    self.create(self.passwd_request_body()?).await
+                    self.create(self.passwd_request_body(pwd)).await
+                }
+                AuthType::OAuth(token) => {
+                    log::info!("Starting session with oauth authentication");
+                    self.create(self.oauth_request_body(token)).await
```

Comment:
Here's where we'd want to use generics on the AuthType trait and implement `get_request_body` for each auth type.